pkg/skaffold/cluster/minikube.go

```diff
@@ -24,21 +24,36 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"sync"
 
+	"github.com/blang/semver"
 	"github.com/sirupsen/logrus"
 	"k8s.io/client-go/util/homedir"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/version"
+)
+
+const (
+	minikubeVersionWithUserFlag = "1.18.0"
 )
 
 var GetClient = getClient
 
 // To override during tests
 var (
-	minikubeBinaryFunc = minikubeBinary
-	getClusterInfo     = context.GetClusterInfo
+	minikubeBinaryFunc    = minikubeBinary
+	getClusterInfo        = context.GetClusterInfo
+	GetCurrentVersionFunc = getCurrentVersion
+
+	findOnce sync.Once
+	mk       = struct {
+		version semver.Version
+		path    string
+		err     error
```

Comment:
I would move err to the top as it determines the validity of the remaining fields.  If `err != nil`, then path and version are unset.  If `err == nil` then path and version should be valid.